src/hexgame/board.rs

```diff
@@ -215,6 +216,47 @@ impl Board {
     pub fn make_move(&mut self, coords: (usize, usize), owner: Ownership) {
         self.cells[coords.0][coords.1].ownership = owner;
     }
+
+    fn encode(&self, coords: (usize, usize)) -> Option<usize> {
+        //
+        // It encodes a coordinate touple into a single integer
+        // so that it can be treated in a general way as RL System
+        //
+        // To explain the mapping we want this 3x4 table:
+        //
+        // 00 01 02 03
+        // 10 11 12 13
+        // 20 21 22 23
+        //
+        //
+        // to be mapped to the following:
+        //
+        // 0  1  2  3
+        // 4  5  6  7
+        // 8  9 10 11
+        //
+        // Then it's easy to spot the formula,
+        // i.e. 7 = 3 + 4
+        //        = 3 + 1*4
+        //        = (3) + 1*4
+        //        = (coords.1)+coords.0*self.dimy
+        //
+
+        if !self.is_valid_square(coords.0.try_into().unwrap(), coords.1.try_into().unwrap()) {
+            None
+        } else {
+            Some((coords.1) + coords.0 * self.dim_y)
+        }
+    }
+
+    fn decode(&self, id: usize) -> Option<(usize, usize)> {
+        // To decode back we do integer division
+        if id > (self.dim_x * self.dim_y - 1) {
```

Comment:
I think we need to also check that `Id` is  not negative here.

Reply:
well,actually I had that check but i removed that because the compiler tell us that is is a useless comparison. I guess usize is an unsigned integer by defintion, so it's all good.